terra/executor/resources.py

```diff
@@ -0,0 +1,395 @@
+'''
+In a multithreaded or multiprocessing environment, it may become necessary to
+divide up and balance a limited "resource" among multiple workers. When a
+:class:`terra.task.TerraTask` is distributed to a worker, each worker might
+need to know what resource id to use, when you have this limited resource.
+
+For example if you have 4 GPUs and each GPU can only handle two simultaneous
```

Comment:
I'd strike this paragraph. Regardless, the end is redundant with the beginning. Simultaneous is also redundant with "at a time"